### PR TITLE
Write install guide

### LINKED ISSUE #688: Make an installation guide in the document
From v1.5, users need to install commands and libraries before they install Chainer. We need to make an official installation guide.

- Anaconda case
- Ubuntu case
- CentOS case

### PR DESCRIPTION
**Pull request by unnonouno**
Fix #688

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+We recommend these platforms.
+
+* `Ubuntu <http://www.ubuntu.com/>`_ 14.04 LTS
+* `CentOS <https://www.centos.org/>`_ 7
+
+Chainer is supported on Python 2.7.6+, 3.4.3+, 3.5.0+.
+Chainer and dependent libraries use these tools and libraries.
+
+* gcc
```
Comment (okuta) [inline docs/source/install.rst]: g++?

File: docs/source/install.rst [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+Chainer depends to these Python packages:
+
+* `Numpy <http://www.numpy.org/>`_ 1.9
+* `Six <https://pythonhosted.org/six/>`_ 1.9
+* `h5py <http://www.h5py.org/>`_ 2.5.0
+
+CUDA support
+
+* `CUDA <https://developer.nvidia.com/cuda-zone>`_ 6.5, 7.0, 7.5
+* `Cython <http://cython.org/>`_ 0.23
```
Comment (okuta) [inline docs/source/install.rst]: Cython is not needed for install of CUDA suppport.

File: docs/source/install.rst [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+
+Install Chainer from source
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use ``setup.py`` to install Chainer from source::
+
+  $ tar zxf chainer-x.x.x.tar.gz
+  $ cd chainer-x.x.x
+  $ python setup.py install
```
Comment (okuta) [inline docs/source/install.rst]: `$ pip install -U cython` is required for install of Chainer.

File: docs/source/install.rst [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+      $ sudo CUDA_PATH=/opt/nvidia/cuda pip install chainer
+
+
+.. _install_cudnn:
+
+Install Chainer with CUDA and cuDNN
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+cuDNN is a library for Deep Neural Networks that nVidia provides.
```
Comment (okuta) [inline docs/source/install.rst]: NVIDIA?

File: docs/source/install.rst [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+Chainer can use cuDNN.
+If you want to enable cuDNN, install cuDNN and CUDA before installing Chainer.
+We recommend you to install cuDNN to CUDA directory::
+
+  $ cp /path/to/cudnn.h $CUDA_PATH/include
+  $ cp /path/to/cudnn.so $CUDA_PATH/lib64
```
Comment (okuta) [inline docs/source/install.rst]: Is there a difference by the user environment?

File: docs/source/install.rst [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+=============
+
+.. _before_install:
+
+Before installing Chainer
+-------------------------
+
+We recommend these platforms.
+
+* `Ubuntu <http://www.ubuntu.com/>`_ 14.04 LTS
```
Comment (okuta) [inline docs/source/install.rst]: 64-bit environment?

File: docs/source/install.rst (hunk 3 above)
Comment (unnonouno) [inline docs/source/install.rst]: I wrote install method for developers

**Comment by unnonouno** [comment]
This patch also fix #713

**Comment by okuta** [comment]
LGTM!

**Pull request merged**